[Preview]

royendo opened the pull request:
Moving Connector and OLAP to Docs under CONNECT 

WIth a lot more UI friendly design! Still working on the actual content
<img width="1875" height="1686" alt="Screenshot 2025-07-17 at 22 17 43" src="https://github.com/user-attachments/assets/7531423b-78d7-4bd5-a6cd-5dcb1605b45e" />

<img width="1288" height="1147" alt="Screenshot 2025-07-17 at 22 24 56" src="https://github.com/user-attachments/assets/b547e977-199d-4ba6-99c0-7612478bd40a" />

Which means no more duplicate source and olap docs, 
**Reference** (as is explained in the other doc is strictly for YAML parameters and not explanations.

**Docs** are more wordy and explanatory of what we can connect to as a concept.

Still a bit of work to do but want to open the Draft PR if anyone interested in seeing designs :)

royendo commented:
notes for tomorrow connect/source/connectors to connect/connector/sources
makes more sense 

write out MotherDuck Doc, and Templating Sources 

make sure that the links and images are actually working!

royendo commented:
opening this PR up, let me know your thoughts,

at some point i mixed up branches and had to pull the other changes to this one so need to recheck once we merge my other two PRs.

This moves a lot of docs around but i ran a custom playwright check for icons and returned green for links, but will need to do a final run 👍

ericpgreen2 reviewed (changes requested):
1. Can we remove a layer of hierarchy here so the list of connectors is 2-layers down, not 3-layers down?
    <img width="300" height="844" alt="image" src="https://github.com/user-attachments/assets/3f95e324-c356-43aa-a1a8-10e6f4b9ec26" />
2. Can we sort the connectors alphabetically? (Helps for scanning a list for what you're looking for.)
3. Optionally, we could have a directory for each "kind" of data source: "Object storage", "Data warehouses", "Databases", "OLAP engines", "Other"
4. Should this section be all about connecting _data_, and we move the Slack connector to the "Integrate" section?
5. Can we have either content or a redirect at `docs.rilldata.com/connect`:
    <img width="1918" height="1009" alt="image" src="https://github.com/user-attachments/assets/69411def-3a4b-447e-ad6c-eee47b2b6bd3" />
6. Whichever approach we take for item 5, can we, near the very top, describe the different connection strategies (e.g. embedded OLAP + ingest your own data, BYO OLAP, BYO OLAP w/ "native" connectors, etc.)
7. These YAML reference links did not take me to the connector-specific section:
    <img width="315" height="252" alt="image" src="https://github.com/user-attachments/assets/f3ba9674-26dc-4e4d-b9d2-76ceb6de0fef" />

ericpgreen2 commented on docs/docs/connect/connect.md:
```diff
@@ -0,0 +1,216 @@
+
+<div className="connector-icon-grid">
+  <ConnectorIcon
+    icon={<img src="/img/connect/icons/Logo-Athena.png" alt="Athena" />}
```
(Like my comment in your other PR), each of these icons should be SVGs not PNGs

royendo replied:
will pull branch changes here

ericpgreen2 commented on docs/docs/connect/connect.md:
```diff
@@ -0,0 +1,216 @@
+
+:::info Full List Of Connectors
+
+Rill is continually adding new sources and connectors in our releases. For a comprehensive list, you can refer to our [Connectors](/connect/) page. Please don't hesitate to [reach out](/contact) either if there's a connector you'd like us to add!
```
This is linking to the current page

ericpgreen2 commented on docs/docs/connect/connector/motherduck.md:
```diff
@@ -2,7 +2,7 @@
 title: DuckDB / MotherDuck
 description: Connect to data in DuckDB locally or MotherDuck
 sidebar_label: DuckDB / MotherDuck
```
This looks like it's out of alphabetical order. What about moving it up to be sorted as a "D"?
<img width="273" height="550" alt="image" src="https://github.com/user-attachments/assets/e87550e8-4fae-45f9-b4f2-3525a45726c2" />

royendo replied:
was sorting on file name [motherduck] woops.

ericpgreen2 reviewed (changes requested):
1. The "Connect" > "Connector" double headings/directories feel like they're too similar...
    <img width="296" height="80" alt="image" src="https://github.com/user-attachments/assets/affddee6-99a5-48f7-9103-5171b0f98abd" />
    ... what if:

    - We move the content from the "Connect" page into the section's first page called something like "Data architectures". This'll make the page easier to find and it'll be a buffer between "Connect" and the "Connector" subheading/directory.
    - Either we should more clearly delineate between Data Sources (rename "Connectors" -> "Data Sources") and "OLAP Engines", or we should put all connectors (both Data Sources & OLAP engines) into the "Connectors" subheader (because they're all "connectors"). 
2. Most of the connector pages end with a bunch of these callout boxes (5 on the Postgres page pictured below). They're a bit overwhelming :) and hard to read when they're stacked like this. Can we find a way to avoid stacking >1 in a row?
    <img width="1913" height="1007" alt="image" src="https://github.com/user-attachments/assets/a444db38-4795-4008-80cd-07be0a1032ce" />
3. Can we remove this from our docs? It'd be best to advertise just the 1 best practice way of setting credentials (the `connector.yaml` files)
    <img width="979" height="137" alt="image" src="https://github.com/user-attachments/assets/a2fa1268-9072-40e7-a84e-8e5f0286663e" />

royendo commented:
some fixes above (not all) will come back tomorrow :)

royendo commented:
not sure why end tests failing but realized we'll need to update all the link to 

```
      href={connector.docsUrl || "https://docs.rilldata.com/build/connect/"}
to
 href={connector.docsUrl || "https://docs.rilldata.com//reference/project-files/connectors#sqlite
```

rill/web-common/src/features/sources/modal/NeedHelpText.svelte

Not sure how docsUrl is getting st (just took a few moments to look at it) but something we'll need to fix on this PR

royendo commented on docs/docs/connect/templating.md:
Sounds good to me. (title/navbar)

I agree they (deploy/tempalating and build/templating) are similar and I think the discussion here is at what step we'd expect a user to start considering dev /prod. We should only have one instance of it.

We've had some back and forth about the discussion to automatically add a dev parittion in source models to ensure good performance and ingesting times on a first experience. 

**The question to ask:**
 Do we expect when a user just starts Rill that they'd need to consider templating at the beginning (they already have deployment in mind) or as an after thought after theyve deployed and need to make changes and things are a bit slow. 
 
 In favor of 1, it makes its an important thing to keep in mind for costs when trying to select * from large_table, but also adds more friction to deployment.
In favor of 2, when a unknowing user tries to project clone, they dont suddenly cost the users $$$ for select * unknowingly (snowflake, bq, etc.)

royendo commented on docs/docs/connect/connect.md:
````diff
@@ -0,0 +1,220 @@
+
+:::
+
+:::note Have a firewall setup?
+You need to whitelist the following IP addresses to connect to/from Rill Cloud and your service behind the firewall.
+```
+35.196.245.100
+34.74.117.37
+35.196.153.31
+34.75.22.143
+34.148.167.51
+35.237.60.193
+```
````
copied the formatting from Olap and seems better (still need to consider this if they've got locked down securities...) not sure where else this might fit

royendo replied:
dont think this would go under source models since its still on conneting stage.

royendo commented on docs/docs/connect/connect.md:
```diff
@@ -0,0 +1,220 @@
+</div>
+
+
+:::tip Avoid Pre-aggregated Metrics
+
+Rill works best for slicing and dicing data meaning keeping data closer to raw to retain that granularity for flexible analysis. When loading data, be careful with adding pre-aggregated metrics like averages as that could lead to unintended results like a sum of an average. Instead, load the two raw metrics and calculate the derived metric in your model or dashboard.
```
could move this to /build/model

royendo commented:
Removed Templating from Deploy and moved to Connect! 

There's a bit more content that I will add in a following PR but wanted to get this one out of door., as its already pretty big. 

Follow ups are:
- #7692 - Splash pages 
- Redirect checker (discussed with Nishant)
- Content Overview (IE: rework the deploy dashboard contents, focus on UI first) 
- Media update (Out of date screenshots and ugly GIFs (made by me )
- Additional Guides

royendo commented:
1. TIL you can do indented tips.

2. an interesting convo to have with product to decide what the best use case of the product is. Last we decided was this 
https://www.figma.com/design/Qt6EyotCBS3V6O31jVhMQ7/RILL-Latest?node-id=19220-1044411&t=bG29z95XAvoAxWDq-1

seems to match. OLAP then Data

ericpgreen2 commented on docs/docs/build/metrics-view/metrics-view.md:
```diff
@@ -21,6 +21,15 @@ Starting in version 0.50, the metrics view has been separated from the dashboard
+
+## Creating Metric Views with AI
+
+In order to streamline the process and get to a dashboard as quickly as possible, we've added the "Create Metrics with AI" and "Create Dashboard with AI" option! This will pass your schmea to OpenAI to suggest measure and dimensions to get started with Rill. You can define you own OpenAI key by creating a [connector file](/reference/project-files/connectors#ai). If you want to disable AI from you ...
```
Grammar: 
- "option" -> "options"
- "schmea" -> "schema"

ericpgreen2 commented on docs/docs/connect/connect.md:
```diff
@@ -0,0 +1,276 @@
+    linkLabel="Learn more"
+    referenceLink="pinot"
+  />
+    <ConnectorIcon
+    icon={<img src="/img/connect/icons/Headset.svg" alt="Pinot" />}
+    content="Don't see an OLAP engine that you're looking for? Chat with us about your use case!"
+    link="/contact"
+    linkLabel="Learn more"
+  />
```
Nit: I wouldn't format this in the same way as the actual connectors. Maybe just a one-liner sentence below would be better. 

Related, it didn't seem consistent to me that only the OLAP engines that this "Contact us" tile, and all the other connector types did not.

ericpgreen2 commented on docs/docs/connect/connect.md:
```diff
@@ -0,0 +1,276 @@
+</div>
+
+
+## Data Warehouse Connectors
```
For consistency and tightness, I think the headers should be: "Data Warehouses", "Databases", "Object storage", "Other data connectors", "Other integrations"

ericpgreen2 commented on docs/static/img/connect/icons/Logo-Snowflake-1.svg:
Is it intentional that there are two Snowflake logos?

ericpgreen2 commented on web-common/src/components/icons/connectors/MicrosoftAzureBlobStorage.svelte:
Double-checking that this edit was intentional? This component is used in the "Add Data" modal and now looks like this:
<img width="568" height="160" alt="image" src="https://github.com/user-attachments/assets/bd1536bb-b649-4f17-b96b-34e6c7efb274" />

ericpgreen2 reviewed (approved):
Approving with a few last comments!
